File: sdk-ts/examples/proxy_smoke.ts
// Smoke example for the TS SDK proxy mode. Routes one fetch through the
// EchoProxy proxy. The orchestrator runs this with a unique tag in the
// path and verifies the event landed in ClickHouse.
//
// Run: ECHOPROXY_API_KEY=sk_test_demo ECHOPROXY_TAG=abc node --experimental-strip-types examples/proxy_smoke.ts

import { fetch } from "../src/proxy.ts";

const tag = process.env.ECHOPROXY_TAG ?? "ts-default";
const target = process.env.ECHOPROXY_EXAMPLE_TARGET ?? "http://upstream-mock:9000";

const url = `${target}/api/users/sdkbench-ts-${tag}`;
const res = await fetch(url);
const body = await res.text();
console.log(`ts sdk: ${url} -> ${res.status} (${body.length} bytes)`);
process.exit(res.status === 200 ? 0 : 1);

File: sdk-python/examples/proxy_smoke.py
"""Smoke example for the Python SDK proxy mode.

Routes one GET through the EchoProxy proxy via echoproxy.proxy.session() and
prints the response status. The orchestrator runs this with a unique tag
in the path and verifies the event landed in ClickHouse.
"""

from __future__ import annotations

import os
import sys

import echoproxy.proxy as sid


def main() -> int:
    tag = os.environ.get("ECHOPROXY_TAG", "py-default")
    target = os.environ.get("ECHOPROXY_EXAMPLE_TARGET", "http://upstream-mock:9000")

    url = f"{target}/api/users/sdkbench-py-{tag}"
    s = sid.session()
    r = s.get(url, timeout=5)
    print(f"py sdk: {url} -> {r.status_code} ({len(r.content)} bytes)")
    return 0 if r.status_code == 200 else 1


if __name__ == "__main__":
    sys.exit(main())
